app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java

```diff
@@ -0,0 +1,199 @@
+        }catch (NullPointerException exception){
+            Log.e(TAG, NULL_VIEWS_EXCEPTION_MESSAGE);
+            Log.e(TAG, exception.toString());
+            }
+    }
+
+    /**
+     * This intercepts the touch event and, by returning false and onTouchEvent(), passes the touchevent
+     * to both itself and its child views (by calling TouchEvent it passes it to itself).
+     * It also detects where the touch was placed so that if the touch is not in the scrollview, the
+     * touch is not passed to the HorizontalScrollView, avoiding the child view swallowing up the long
+     * click. False is passed to still allow MenuItemClick to happen.
+     * However, if the action is ACTION_MOVE, it cancels the touch event in itself and
+     * only gives it to its children, which, in this case is a HorizontalScrollView.
+     *
+     * @param e the intercepted touch event
+     * @return If this function returns true, the MotionEvent will be intercepted,
+     * meaning it will be not be passed on to the child, but rather to the onTouchEvent of this View.
+     */
+    @Override
+    public boolean onInterceptTouchEvent(MotionEvent e) {
+        int x = Math.round(e.getRawX());
+        int y = Math.round(e.getRawY());
+        try {
+            scrollView.getGlobalVisibleRect(scrollViewRect);
+
+            boolean touchedScrollView =
+                    x > scrollViewRect.left && x < scrollViewRect.right &&
+                            y > scrollViewRect.top && y < scrollViewRect.bottom;
+
+            boolean emptyTruncateText = songTruncated.isEmpty();
+            boolean isTextTruncated = songTruncated.endsWith("…");
+
+            switch (e.getAction()) {
+                case MotionEvent.ACTION_DOWN:
+                    if (!touchedScrollView || (!emptyTruncateText && !isTextTruncated)){
+                        scrollView.cancelPendingInputEvents();
+                        return false;
+                    }
+
+                    startX = e.getX();
+                    startY = e.getY();
+                    isTap = true;
+                    onTouchEvent(e);
+
+                    break;
+
+                case MotionEvent.ACTION_MOVE:
+                    float distanceY = Math.abs(e.getY() - startY);
+                    if (touchedScrollView) {
+                        float distance = Math.abs(e.getX() - startX);
+
+                        // Scrolling the view: cancel event to prevent long press
+                        if (distance > MAX_CLICK_DISTANCE) {
+                            if (!emptyTruncateText){
+                                textView.setText(song);
+                            }else{
+                                Log.e("E/TouchInterceptFL","songTruncated is empty or null. Did you remember " +
+                                        "to set the song string when setting the song name in your text view?");
+                            }
+                            cancelPendingInputEvents();
```

Comment:
This chunk of code is identical to the block of code for handling vertical scrolling below. Might be better to just pull it out into a private method or something.